lib/matplotlib/backend_bases.py

```diff
@@ -3428,7 +3422,13 @@ def add_tool(self, name, tool, group=None, position=None):
                           'not added')
             return
 
-        self._tools[name] = tool_cls(self, name)
```

Comment:
Navigation is not really optional because the user is not supposed to instantiate the tool by himself, it's navigation that does it.
I don't thing we should plan for something like that, if that starts to happen, then we can raise an error in the `__init__` method of the tools